src/Sdk/RSWebApi/Contracts/AcquireJobRequest.cs

```diff
@@ -7,7 +7,29 @@ namespace GitHub.Actions.RunService.WebApi
     [DataContract]
     public class AcquireJobRequest
     {
+        [DataMember(Name = "jobMessageId", EmitDefaultValue = false)]
+        public string JobMessageID { get; set; }
+
+        // The earliest versions of this type serialized the JobMessageID under a field named "streamId"
+        // This private property is expected to be a short-lived shim to aid in transitioning over to the new schema.
         [DataMember(Name = "streamId", EmitDefaultValue = false)]
-        public string StreamID { get; set; }
+        private string DeprecatedJobMessageIdAlias
+        {
+            get
+            {
+                return this.JobMessageID;
+            }
+            set
+            {
+                // Note that this is a private property, so the only way to reach this setter is via deserialization.
+                // Deserializtion is NOT an expected use case on the Runner side as of early 2023.
+                // In the unlikely event that deserializtion does become a valid use case, we include this guard
+                // to ensure that a valid JobMessageID can't be overwritten by this otherwise unreachable code path.
+                if (string.IsNullOrEmpty(this.JobMessageID))
+                {
+                    this.JobMessageID = value;
+                }
```

Comment:
It's worth pointing out here that the main factor at play in this unlikely scenario is the order in which the fields are deserialized.  Experimentation reveals it depends more on the raw JSON order than anything else.  Setting the `Order` property in the `DataMember` attribute did not produce a reliable guarantee that one field would always be deserialized before another.

Reply:
I also considered just making this setter a `throw` since we don't anticipate this code path to ever be exercised.